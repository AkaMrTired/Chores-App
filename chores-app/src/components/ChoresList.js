import React from "react";

import PropTypes from "prop-types";

const ChoresList = ({ availableChores }) => {
  return (
    <div className="container">
      <h1>Chores</h1>
      <button type="button">
        <a href="/addchore">Add new chore +</a>
        {/* mapping function to go through the chores list and render them. */}
      </button>
      <p>{availableChores}</p>
    </div>
  );
};

// need to update proptypes when we know what the data looks like
ChoresList.propTypes = {
  availableChores: PropTypes.arrayOf(PropTypes.text).isRequired,
};
export default ChoresList;
