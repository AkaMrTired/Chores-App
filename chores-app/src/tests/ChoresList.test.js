import {
  render, // screen
} from "@testing-library/react";
import React from "react";
import ChoresList from "../components/ChoresList";

describe("the homepage", () => {
  it("matches the snapshot", () => {
    const { asFragment } = render(<ChoresList />);

    expect(asFragment()).toMatchSnapshot();
  });
  it("should render basic fields", () => {
    render(<ChoresList />);
    //     expect(screen.getByText(/home page/i)).toBeInTheDocument();
    //     expect(screen.getByText(/email/i)).toBeInTheDocument();
    //     expect(screen.getByRole("button", { name: /Log In/i })).toBeInTheDocument();
    //     expect(screen.getByText(/password/i)).toBeInTheDocument();
    //     expect(screen.getByText(/not yet a member/i)).toBeInTheDocument();
    //     expect(
    //       screen.getByRole("button", { name: /Sign Up/i })
    //     ).toBeInTheDocument();
  });
});
