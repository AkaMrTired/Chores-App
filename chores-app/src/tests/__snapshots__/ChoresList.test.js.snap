// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the homepage matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <h1>
      Chores
    </h1>
    <button
      type="button"
    >
      <a
        href="/addchore"
      >
        Add new chore +
      </a>
    </button>
    <p />
  </div>
</DocumentFragment>
`;
